eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -7,7 +7,7 @@
       "OSConfiguration": {
         "macos-latest": {
           "OSVmImage": "env:MACVMIMAGE",
-          "XCODE_VERSION": "15.4"
+          "XCODE_VERSION": "16.4"
```

Comment:
16.4 is the default https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md#xcode

It might be the right thing to remove this line entirely unless you want to pin for backwards compatibility testing.